modules/improvedigitalBidAdapter.js

```diff
@@ -10,6 +10,70 @@ const BIDDER_CODE = 'improvedigital';
 const RENDERER_URL = 'https://acdn.adnxs.com/video/outstream/ANOutstreamVideo.js';
 const VIDEO_TARGETING = ['skip', 'skipmin', 'skipafter'];
 
+class Razr {
+  static addBidData(data) {
+    const {bid, bidRequest} = data || {};
+
+    if (this.isValidBid(bid)) {
+      const rendererConfig = mergeDeep(
+        {},
+        config.getConfig('improvedigital.rendererConfig'),
+        deepAccess(bidRequest, 'params.rendererConfig')
+      );
+
+      this.bids = this.bids || {};
+      this.bids[bid.requestId] = {
+        ...data,
+        adm: bid.ad,
+        config: rendererConfig
+      };
+
+      bid.ad = `<script>window.top.postMessage({razrBidId: "${bid.requestId}"}, "*");</script>`;
+      this.addListenerOnce();
+    }
+  }
+
+  static isValidBid(bid) {
+    return bid && /razr:\\?\/\\?\//.test(bid.ad);
+  }
+
+  static render(bidId, event) {
+    const ns = window.razr = window.razr || {};
+    ns.queue = ns.queue || [];
+
+    ns.queue.push({
+      ...this.bids[bidId],
+      type: 'prebid',
+      event
+    });
+
+    if (!this.loaded) {
+      const s = document.createElement('script');
+      s.type = 'text/javascript';
+      s.async = true;
+      s.src = 'https://razr.improvedigital.com/renderer.js';
+
+      const x = document.getElementsByTagName('script')[0];
+      x.parentNode.insertBefore(s, x);
```

Comment:
please use ajax methods

Reply:
The script to be loaded will add logic to the page. Can you please elaborate on how and why this should be done with ajax?

Reply:
because it is a prebid rule

Reply:
you can use ajax like this https://github.com/prebid/Prebid.js/pull/7699/files or you can import the loadExternalScript function from utils. You cannot you the logic here.

Reply:
Thank you. We'll review the options.

Reply:
ty; I noted on the main thread, the logic in the oneVideo adapter would likely work for you as well.

Reply:
code updated to rely on `executeRenderer`